enhancements/declarative-index-config.md

```diff
@@ -135,27 +130,32 @@ $ cat etcd.json
     "properties":[
         {
             "name": "olm.package",
+            "type": "required",
             "values": {
-                "packageName": "etcd",
-                "version": "0.9.0"
+                "packageName": "other-operator",
+                "version": ">=1.2.0 <2.0.0-0"
             }
         },
         {
             "name":"olm.gvk",
             "type": "provided",
             "values":{
-                "group": "etcd.database.coreos.com", 
-                "kind": "EtcdBackup", 
+                "group": "etcd.database.coreos.com",
+                "kind": "EtcdBackup",
                 "version": "v1beta2"
             }
         },
         {
             "name": "olm.channel",
-            "value": "singlenamespace-alpha"
-        }
+            "values": {
```

Comment:
The main reason for this was to move "replaces" into "values". From both a UX and a parsing perspective, IMO its easier to document and remember:
1. "name": "olm.channel" always uses "values"
2. "values"."name" is always where I should find/put the channel name
3. "values"."replaces" is where I can optionally say what bundle this bundle replaces in this channel.

Reply:
Ah! I just found https://github.com/operator-framework/operator-registry/blob/92a4c92168862d509841cef6fa949a1b6dec1ec9/pkg/registry/types.go#L186

Are you suggesting basically reusing that exact type (or a re-definition of it) for declarative config properties, such that we have `"value"` and it can contain any valid json data?

Reply:
Updated based on this feedback. After re-reading your comment and finding examples of other OLM property types, I think I have this aligned with what you're imagining. Let me know!

Reply:
yup, this is what I was expecting! thanks